src/libraries/Native/Unix/System.Security.Cryptography.Native/openssl.c

```diff
@@ -1311,6 +1315,8 @@ int32_t CryptoNative_EnsureOpenSslInitialized()
     // If 1.0, call the 1.0 one.
     // Otherwise call the 1.1 one.
 #ifdef FEATURE_DISTRO_AGNOSTIC_SSL
+    InitializeOpenSSLShim();
```

Comment:
The semantics of the containing method is that it can be called multiple times (thus the name starting with Ensure). The EnsureOpenSsl10Initialized / EnsureOpenSsl11Initialized  both work that way:
https://github.com/dotnet/runtime/blob/1d9e50cb4735df46d3de0cee5791e97295eaf588/src/libraries/Native/Unix/System.Security.Cryptography.Native/openssl.c#L1179-L1185
So we should make the InitializeOpenSSLShim behave the same way or maybe just put the InitializeOpenSSLShim call into those functions, which would seem better.

Reply:
Agreed. I will try moving initialization call into these functions.